### PR TITLE
type declaration base

### PR DESCRIPTION
**Pull request by robjweiss**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    plugins: Array<Plugin>
+}
+
+export class Client extends EventEmitter {
+    public plugins: Array<Plugin>
+
+    constructor(rabbitmqURL: string, options: Options)
+
+    start(): this
+    close(): this
```
Comment (robjweiss) [inline index.d.ts:72]: I'm not so sure about the return values of these functions

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    DefaultOptions: typeof DefaultOptions,
+    Base: typeof Plugin
+}
+
+/**
+ * Client
+ */
+
+type Options = {
+    logger: bunyan
```
Comment (naggingant) [inline index.d.ts:62]: I think we can declare an interface with `debug` `info` `warn` `error` that is `(s: string) => void` or `(v: any) => void` (whichever works) and not include bunyan in the dependency

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import { type EventEmitter } from 'events'
+import type bunyan from 'bunyan'
+
+export as namespace hato;
+
+/**
+ * Plugins
+ */
```
Comment (naggingant) [inline index.d.ts:8]: I think I will move these definitions to the plugin dir later. JFYI

**Review by naggingant — COMMENTED** [review]
Thank you so much
Just one comment that I would like to conclude before publishing

File: index.d.ts (hunk 3 above)
Comment (robjweiss) [reply index.d.ts:7]: Ok, I think that's fine :+1:

File: index.d.ts (hunk 2 above)
Comment (robjweiss) [reply index.d.ts]: Sure, I did this.

File: index.d.ts (hunk 1 above)
Comment (naggingant) [reply index.d.ts:78]: I think that's correct for `start()`, and `close()` is probably `Promise<void>`

**Review by PuKoren — APPROVED** [review]
:+1:

**Comment by naggingant** [comment]
Test from the other plugins. Publishing

**Pull request merged**